src/ContainerIntegration.jsx

```diff
@@ -10,25 +10,23 @@ import { EmptyStatePanel } from "cockpit-components-empty-state.jsx";
 
 const _ = cockpit.gettext;
 
-export const renderContainerPublishedPorts = (ports) => {
+// ports is a mapping like { "5000/tcp": [{"HostIp": "", "HostPort": "6000"}] }
+export const renderContainerPublishedPorts = ports => {
     if (!ports)
         return null;
 
-    const result = ports.map(port => {
-        // podman v4 has different names than v3
-        const protocol = port.protocol;
-        const hostPort = port.hostPort || port.host_port;
-        const containerPort = port.containerPort || port.container_port;
-        const hostIp = port.hostIP || port.host_ip || '0.0.0.0';
-
-        return (
-            <ListItem key={ protocol + hostPort + containerPort }>
-                { hostIp }:{ hostPort } &rarr; { containerPort }/{ protocol }
-            </ListItem>
-        );
+    const items = [];
+    Object.entries(ports).forEach(([containerPort, hostBindings]) => {
+        hostBindings.forEach(binding => {
```

Comment:
I managed to get error:
```
Uncaught TypeError: hostBindings is null
    renderContainerPublishedPorts ContainerIntegration.jsx:20
    renderContainerPublishedPorts ContainerIntegration.jsx:19
    ContainerIntegration ContainerIntegration.jsx:97
    React 11
```

These are my ports:
```
               "Ports": {
                    "3400/tcp": null,
                    "8080/tcp": [
                         {
                              "HostIp": "",
                              "HostPort": "3404"
                         }
                    ]
               },

```

I am actually not sure how I created this :man_shrugging:

Reply:
Eww -- Not sure either, but let's not crash the UI on it. Fixed.